plugins/modules/bootc_manage.py

```diff
@@ -0,0 +1,98 @@
+# or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: bootc_manage
+author:
+- Ryan Cook (@cooktheryan)
+short_description: Bootc Switch and Upgrade
+description:
+    - This is a bootc module that manages the switching and upgrading of bootc.
+options:
+    state:
+        description:
+            - 'Control to apply the latest image or switch the image.'
+            - 'This is a string value.'
+            - 'NOTE: This will not reboot the system.'
+            - 'Please use the reboot module to reboot the system.'
+        required: true
+        type: str
+        choices: ['switch', 'latest']
+    image:
+        description:
+            - 'The image to switch to.'
+            - 'This is required when state is ''switch''.'
+        required: false
+        type: str
+
+'''
+
+EXAMPLES = '''
+# Switch to a different image
+- name: Provide image to switch to a different image and retain the current running image
+  community.bootc_manage:
+    state: switch
+    image: "example.com/image:latest"
+
+# Apply updates of the current running image
+- name: Apply updates of the current running image
+  community.bootc_manage:
+    state: latest
+'''
+
+RETURN = '''
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = dict(
+        state=dict(type='str', required=True, choices=['switch', 'latest']),
+        image=dict(type='str', required=False)
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    state = module.params['state']
+    image = module.params['image']
+
+    if state == 'switch':
+        if not image:
+            module.fail_json(msg="Image is required when state is 'switch'")
+        command = 'bootc switch {} --retain'.format(image)
+        rc, out, err = module.run_command(command)
+        if 'Queued for next boot: ' in out:
+            result = {'changed': True, 'output': out}
+            module.exit_json(**result)
+        elif 'Image specification is unchanged.' in out:
+            result = {'changed': False, 'stderr': out}
+            module.exit_json(**result)
+        else:
+            result = {'changed': False, 'stderr': err}
+            module.fail_json(msg='ERROR: Failed to switch image.', **result)
+
+    elif state == 'latest':
+        command = 'bootc upgrade'
+        rc, out, err = module.run_command(command)
+        if rc == 0 and 'No changes in ' in out:
+            result = {'changed': False, 'output': out}
+            module.exit_json(**result)
+        elif rc == 0 and 'Queued for next boot: ' in out:
+            result = {'changed': True, 'output': out}
+            module.exit_json(**result)
+        else:
+            result = {'changed': False, 'stderr': err}
+            module.fail_json(msg='ERROR: Failed to apply updates.', **result)
```

Comment:
This block is the same for both states. Maybe simplify the code not to repeat itself?